viewing_party/party.py

```diff
@@ -0,0 +1,179 @@
+                    friends_unique_watched_list.append(movie_dict)
+                    friend_watched_movie_titles.append(movie_dict["title"])
+
+    # return the list of movie dictionaries representing movies only
+    # the user has watched
+    return friends_unique_watched_list
+
+def get_available_recs(user_data):
+    # create a list of movie dictionaries for movies that the user 
+    # has not seen but a friend has seen, and that the user has access 
+    # to aka the movie's host is in the user's subscriptions list 
+    movie_recs = []
+
+    # iterate through movies that only the user's friends have seen
+    # check if the user has access to the movie's host
+    # check if the movie has already been added to the rec list
+    # if passes both checks add the movie to the rec list
+    for movie_dict in get_friends_unique_watched(user_data):
+        if movie_dict["host"] in user_data["subscriptions"]: 
+                movie_recs.append(movie_dict)
+
+    # return list of dictionaries reccomending movies
+    return movie_recs
+
+def get_new_rec_by_genre(user_data):
+    # create list of movie dictionaries of movies that the user
+    # has not seen, but a friend has, that are in the user's top
+    # genre
+    recs_by_genre_list = []
+
+    # iterate through movies that the user has not seen but 
+    # friends have seen
+    # check if the movie's genre is the user's top genre
+    # if the movie passes the check add it to the recs_by_genre_list
+    for movie_dict in get_friends_unique_watched(user_data):
+        if movie_dict["genre"] == get_most_watched_genre(user_data):
+            recs_by_genre_list.append(movie_dict)
+    
+    # return list of movie recs
+    return recs_by_genre_list
+
+def get_rec_from_favorites(user_data):
+    # create a list of movie dictionaries of the user's fav
+    # movies. aka movies in the user's "favorites" list that
+    # none of their friends have watched
+    recs_from_favorites_list = []
+
+    # iterate through the user's favorite movies
+    # check to see if any of the user's friends have seen it
+    # if a movie passes the check add it to the recs_from_favorites_list
+    for movie_dict in user_data["favorites"]:
+        if movie_dict in get_unique_watched(user_data):
+            recs_from_favorites_list.append(movie_dict)
+    
+    #return list of movie dictionaries for user's fav movie recs
+    return recs_from_favorites_list
+
+# ************************************************************* ~
+# pytest tests/test_wave_01.py
+# pytest tests/test_wave_02.py
+# pytest tests/test_wave_03.py
+# pytest tests/test_wave_04.py
+# pytest tests/test_wave_05.py
+
+# When you need to save changes and add them to github:
+# Add all changed files  git add .
+# Commit changes git commit -m "useful message about the change"
+# Push changed files to github git push
```

Comment:
💡 Quick Tip:  To make your project neater and follow pep8 guidelines, you can have VS Code autoformat your files by right-clicking on a file in VS Code and select Format Document With... then select python from the drop-down menu. You'll then be prompted to install autopep8, click yes and the package will be installed within your virtual environment.